internal/app/crypto.go

```diff
@@ -0,0 +1,52 @@
+package app
```

Comment:
It would be beneficial for all to have test cases (unit tests) as well. This way, no one need to run entire application to be sure whether it is working or not. @yusufpapurcu
Just a suggestion ;) 
Well done, you did a great job, by the way ;)